Run from Container guide V1

DGuhr opened the pull request:
Closes #9465

DGuhr committed:
Run from Container guide V1

Closes #9465

Co-authored-by: Stian Thorgersen <stian@redhat.com>

DGuhr reviewed (commented):
Also:
- fix typo and reference to properties file in enabletls guide
- rename proxy.adoc to reverseproxy.adoc, more concise

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+To start the image, run:
+[source, bash]
+----
+podman|docker run --rm -it --name optimized_keycloak -p 8443:8443 prebuilt_keycloak
```
`-rm` and `-it` aren't needed here

```suggestion
podman|docker run --name optimized_keycloak -p 8443:8443 prebuilt_keycloak
```

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+INFO  [org.key.com.Profile] (main) Preview feature enabled: token_exchange
+----
+stating that the desired feature is enabled.
+Also, opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
```
Probably better to avoid `Also,`
```suggestion

Opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
```

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+----
+INFO  [org.key.com.Profile] (main) Preview feature enabled: token_exchange
+----
+stating that the desired feature is enabled.
```
```suggestion
This shows the desired feature is enabled.
```

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+Also, opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
+
+== Try Keycloak out in development mode
+The easiest way to try out Keycloak from a container for development or testing purposes is using the Development mode(_todo_link_to_missing_guide_) by using the `start-dev` command:
```
I think we agreed not to do a developer guide, and rather just have that in the configuration guide?

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+----
+stating that the desired feature is enabled.
+Also, opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
+
```
Probably worth mentioning that only https is enabled in production mode, and point folks to open `https://localhost:8443`

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+
+[source,bash]
+----
+podman|docker run --rm -it --name keycloak_test -p 8080:8080 \
```
`-rm -it` is not needed

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+
+[source, bash]
+----
+podman|docker run --rm -it --name keycloak_auto_build -p 8080:8080 \
```
I'd say we should rather favour environment variables as that is more the standard container way?

DGuhr replied:
Wanted to give one example to show that you _can_ use the cli commands too if you wanted to. But not hard on that, I could also change this to environment variables. wdyt?

DGuhr replied:
decided to leave as is for now.

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+title="Running in a container"
+summary="Learn how to run Keycloak from a container image"
+priority=20
+includedOptions="">
```
Should we include some options here? Standard stuff that is needed when running a container? Or, at least include the options that you've used in the examples?

DGuhr replied:
I can do this, but might grow quite big. Let me try. Btw: What about KEYCLOAK_ADMIN / KEYCLOAK_ADMIN_PASSWORD - they're not documented yet in the all-config guide, and are kinda special.

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,106 @@
+
+Running the above will spin up a Keycloak server which detects and applies the build configuration first. In the example, it's  `--db=postgres --features-token_exchange=enabled` to set the used database vendor to postgres and enable the token exchange feature.
+
+Keycloak will then start up and apply the configuration for the specific environment. This usually takes way more time than spinning up an already optimized image, and is therefore not recommended.
```
What about something like:

```suggestion
Keycloak will then start up and apply the configuration for the specific environment. This approach results in a significantly bigger startup time, and an image that is mutable, and is therefor not the best practice.
```

stianst reviewed (changes requested):
See comments

DGuhr commented:
@stianst Thanks for the review. New version up, squashed commits and incorporated your feedback. Also changed one example to use features=token_exchange instead of old features-token_exchange. 

Could you pls re-check?

stianst commented on docs/guides/src/main/server/run-containers.adoc:
```diff
@@ -0,0 +1,107 @@
+<#import "/templates/guide.adoc" as tmpl>
```
Rename to containers.adoc for nicer link on the website

DGuhr replied:
done
